packages/remirror__extension-markdown/src/html-to-markdown.ts

```diff
@@ -75,10 +75,18 @@ function cell(content: string, node: Node) {
 const turndownService = new TurndownService({ codeBlockStyle: 'fenced', headingStyle: 'atx' })
   .addRule('taskListItems', {
     filter: (node) => {
-      return (node as HTMLInputElement).type === 'checkbox' && node.parentNode?.nodeName === 'LI';
+      return node.nodeName === 'LI' && node.hasAttribute('data-checkbox');
     },
-    replacement: (_, node) => {
-      return `${(node as HTMLInputElement).checked ? '[x]' : '[ ]'} `;
+    replacement: (content, node) => {
+      const trimmedContent = content
+        .replace(/^\n+/, '') // remove leading newlines
+        .replace(/\n+$/, '\n') // replace trailing newlines with just a single one
+        .replace(/\n/gm, '\n    '); // indent
+
+      const input = childSelector(node as Element, 'input[type=checkbox]') as HTMLInputElement;
```

Comment:
`childSelector` is util I've added that is akin to DOM API `querySelector` - however it only checks the immediate children of the given node.